### PR TITLE
Proxied maintenance request - add gui-apps/swaync as second notification-daemon for wayland

### PR DESCRIPTION
**Pull request by freijon**
As discussed [here](https://github.com/gentoo/guru/commit/1b57866cff7ec63e20e874d2e7551e9ba21d0358), the goal of this PR is to introduce a second notification daemon for wayland after `mako` and move `swaync` from GURU to ::gentoo.

After a discussion with @joecool1029 I'd be willing to become a proxied maintainer to continue maintaining `swaync` with him as co-maintainer. Furthermore, I used to maintain `wezterm` on GURU which was recently added to ::gentoo. I offer to continue (proxied) maintenance of this package if I'm accepted.

Please let me know your thoughts about this.

### CODE REVIEW

File: gui-apps/swaync/swaync-0.7.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+src_prepare() {
+	default
+	vala_setup
+}
+
+src_configure() {
+	local emesonargs=(
+		$(meson_use man man-pages)
+		$(meson_use scripting)
```
Comment (Kangie) [inline gui-apps/swaync/swaync-0.7.3.ebuild:45]: Upstream's README seems to suggest that it's built by default; is there any reason that this shouldn't be enabled unconditionally? Why would a user not want scripting support?

**Comment by Kangie** [comment]
I'd like to see a patch for the meson build warnings, though it isn't essential:

```
WARNING: Project specifies a minimum meson_version '>= 0.50.0' but uses features which were added in newer versions:
 * 0.54.0: {'message with more than one argument'}
 * 0.59.0: {'feed arg in custom_target'}
NOTICE: Future-deprecated features used:
 * 0.56.0: {'dependency.get_pkgconfig_variable'}
```

That should be trivial to fix and submit upstream.

I also note that this won't build with Clang 16:
[swaync-0.7.3-clang16-build.clean.log](https://github.com/gentoo/gentoo/files/10532312/swaync-0.7.3-clang16-build.clean.log)

As the release of the new Clang is imminent I'm not keen on introducing new "Modern C" bugs. The [wiki](https://wiki.gentoo.org/wiki/Modern_C_porting) has a good article on resolving this issue.

Outside of that It's an interesting package, thanks for your work so far!

File: gui-apps/swaync/swaync-0.7.3.ebuild (hunk 1 above)
Comment (freijon) [reply gui-apps/swaync/swaync-0.7.3.ebuild:45]: You mean I should completely remove the USE flag? I agree that it probably should be enabled by default. Maybe some users don't need this feature and would prefer to build a more "minimal" version, so I figured they should be able to disable it.

Comment (Kangie) [reply gui-apps/swaync/swaync-0.7.3.ebuild:45]: I'm thinking that; although choice is nice sometimes as a package maintainer you can make an executive decision.

In this case you could go either way - A user *might* reasonably want to disable scripting entirely, however it is given a section in the README and seems like a major feature. I don't use the software so I'll leave the decision entirely to you :)

**Comment by freijon** [comment]
> I'd like to see a patch for the meson build warnings, though it isn't essential:

I'll create a patch
> 
> ```
> WARNING: Project specifies a minimum meson_version '>= 0.50.0' but uses features which were added in newer versions:
>  * 0.54.0: {'message with more than one argument'}
>  * 0.59.0: {'feed arg in custom_target'}
> NOTICE: Future-deprecated features used:
>  * 0.56.0: {'dependency.get_pkgconfig_variable'}
> ```

would you mind sharing what tool you used to get this warning message? looks like helpful information.
> That should be trivial to fix and submit upstream.

I submitted a PR to upstream
> 
> I also note that this won't build with Clang 16: [swaync-0.7.3-clang16-build.clean.log](https://github.com/gentoo/gentoo/files/10532312/swaync-0.7.3-clang16-build.clean.log)
> 
> As the release of the new Clang is imminent I'm not keen on introducing new "Modern C" bugs. The [wiki](https://wiki.gentoo.org/wiki/Modern_C_porting) has a good article on resolving this issue.
> 
> Outside of that It's an interesting package, thanks for your work so far!

I read the [Modern C porting](https://wiki.gentoo.org/wiki/Modern_C_porting) and [fixing clang16 errors](https://wiki.gentoo.org/wiki/User:Schievel/fixing_clang16_errors) article, but I'm a bit lost how to fix this. Keep in mind that the software is written in Vala. Can I impact how Vala converts its code to C and make the output Clang16 compatible?

**Comment by freijon** [comment]
I'm sorry, I think I messed up with git (I'm not very familiar with pull requests). I will close this one and create a new one.

**Pull request closed without merging**